packages/fetchai/skills/erc1155_deploy/handlers.py

```diff
@@ -0,0 +1,203 @@
+
+        # handle message
+        if fipa_msg.performative == FIPAMessage.Performative.CFP:
+            self._handle_cfp(fipa_msg)
+        elif fipa_msg.performative == FIPAMessage.Performative.INFORM:
+            self._handle_inform(fipa_msg)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def _handle_cfp(self, msg: FIPAMessage) -> None:
+        """
+        Handle the CFP.
+
+        If the CFP matches the supplied services then send a PROPOSE, otherwise send a DECLINE.
+
+        :param msg: the message
+        :return: None
+        """
+        new_message_id = msg.message_id + 1
+        new_target = msg.message_id
+        self.context.logger.info(
+            "[{}]: received CFP from sender={}".format(
+                self.context.agent_name, msg.counterparty[-5:]
+            )
+        )
+
+        contract = self.context.contracts.erc1155
+        self.context.logger.info(contract.item_ids)
+        info_msg = FIPAMessage(
+            message_id=new_message_id,
+            dialogue_reference=("", ""),
+            target=new_target,
+            performative=FIPAMessage.Performative.INFORM,
+            info={
+                "contract": contract.instance.address,
+                "item_ids": contract.item_ids,
+                "trade_nonce": contract.generate_trade_nonce(
+                    self.context.agent_address
+                ),
+            },
+        )
+
+        self.context.outbox.put_message(
+            to=msg.counterparty,
+            sender=self.context.agent_address,
+            protocol_id=FIPAMessage.protocol_id,
+            message=FIPASerializer().encode(info_msg),
+        )
+
+    def _handle_inform(self, msg: FIPAMessage) -> None:
+        """
+        Handle the INFORM.
+
+        If the INFORM message contains the transaction_digest then verify that it is settled, otherwise do nothing.
+        If the transaction is settled, send the data, otherwise do nothing.
+
+        :param msg: the message
+        :return: None
+        """
+        self.context.logger.info(
+            "[{}]: received INFORM from sender={}".format(
+                self.context.agent_name, msg.counterparty[-5:]
+            )
+        )
+        self.context.logger.info(self.context.ledger_apis.apis.get("ethereum"))
+        contract = self.context.contracts.erc1155
+        tx = contract.get_atomic_swap_single_proposal(
+            from_address=self.context.agent_address,
+            to_address=msg.counterparty,
+            item_id=contract.item_ids[0],
+            from_supply=10,
+            to_supply=0,
+            value=0,
+            trade_nonce=500,
+            ledger_api=self.context.ledger_apis.apis.get("ethereum"),
+            skill_callback_id=self.context.skill_id,
+            signature=msg.info.get("signature"),
+        )
+
+        self.context.decision_maker_message_queue.put(tx)
```

Comment:
Creating the atomic swap tx and queuing it to the Decision Maker.

Reply:
All these magic values..really not ideal. Please comb through your code and make it configurable :)